// powerbi-report-component/lib/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Embed = _interopRequireDefault(require("./Embed"));

var _config = require("./config");

var _onEmbedHandlers = require("./onEmbedHandlers");

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var Report = /*#__PURE__*/function (_PureComponent) {
  (0, _inherits2["default"])(Report, _PureComponent);

  var _super = _createSuper(Report);

  function Report(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, Report);
    _this = _super.call(this, props);
    _this.state = {
      currentConfig: null
    };
    _this.performOnEmbed = _this.performOnEmbed.bind((0, _assertThisInitialized2["default"])(_this));
    _this.updateState = _this.updateState.bind((0, _assertThisInitialized2["default"])(_this));
    return _this;
  }

  (0, _createClass2["default"])(Report, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.updateState(this.props);
    }
  }, {
    key: "performOnEmbed",
    value: function performOnEmbed(report, reportRef) {
      var _this$props = this.props,
          embedType = _this$props.embedType,
          reportMode = _this$props.reportMode;

      switch (embedType) {
        case 'report':
          (0, _onEmbedHandlers.reportHandler)(report, reportMode, this.props);
          break;

        case 'dashboard':
          (0, _onEmbedHandlers.dashboardHandler)(report, reportRef, this.props);
          break;

        case 'tile':
          (0, _onEmbedHandlers.tileHandler)(report, this.props);
          break;

        default:
          break;
      }
    }
  }, {
    key: "updateState",
    value: function updateState(props) {
      this.setState({
        currentConfig: (0, _config.createConfig)(props)
      });
    }
  }, {
    key: "render",
    value: function render() {
      if (!this.state.currentConfig) {
        return /*#__PURE__*/_react["default"].createElement("div", null, " Error ");
      }

      return /*#__PURE__*/_react["default"].createElement(_Embed["default"], {
        config: this.state.currentConfig,
        performOnEmbed: this.performOnEmbed,
        style: this.props.style
      });
    }
  }], [{
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps(props) {
      return {
        currentConfig: (0, _config.createConfig)(props)
      };
    }
  }]);
  return Report;
}(_react.PureComponent);

Report.propTypes = {
  embedType: _propTypes["default"].string.isRequired,
  tokenType: _propTypes["default"].string.isRequired,
  accessToken: _propTypes["default"].string.isRequired,
  embedUrl: _propTypes["default"].string.isRequired,
  embedId: _propTypes["default"].string,
  pageName: _propTypes["default"].string,
  extraSettings: _propTypes["default"].object,
  permissions: _propTypes["default"].string,
  onLoad: _propTypes["default"].func,
  onError: _propTypes["default"].func,
  onSelectData: _propTypes["default"].func,
  onPageChange: _propTypes["default"].func,
  onTileClicked: _propTypes["default"].func,
  style: _propTypes["default"].object,
  reportMode: _propTypes["default"].string,
  datasetId: _propTypes["default"].string,
  titleId: _propTypes["default"].string
};
var _default = Report;
exports["default"] = _default;
