// office-ui-fabric-react/lib-commonjs/components/Fabric/Fabric.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Styling_1 = require("../../Styling");
var inheritFont = { fontFamily: 'inherit' };
var GlobalClassNames = {
    root: 'ms-Fabric',
    bodyThemed: 'ms-Fabric-bodyThemed'
};
exports.getStyles = function (props) {
    var theme = props.theme, className = props.className, isFocusVisible = props.isFocusVisible, applyTheme = props.applyTheme;
    var classNames = Styling_1.getGlobalClassNames(GlobalClassNames, theme);
    var focusVisibility = isFocusVisible ? 'Visible' : 'Hidden';
    return {
        root: [
            classNames.root,
            // keywords for search: is-focusVisible ms-Fabric--isFocusVisible ms-Fabric--isFocusHidden
            "is-focus" + focusVisibility + " ms-Fabric--isFocus" + focusVisibility,
            theme.fonts.medium,
            {
                color: theme.palette.neutralPrimary,
                selectors: {
                    '& button': inheritFont,
                    '& input': inheritFont,
                    '& textarea': inheritFont
                }
            },
            // apply theme to only if applyTheme is true
            applyTheme && {
                color: theme.semanticColors.bodyText,
                backgroundColor: theme.semanticColors.bodyBackground
            },
            className
        ],
        bodyThemed: [
            {
                backgroundColor: theme.semanticColors.bodyBackground
            }
        ]
    };
};
